server/infinispan/src/main/java/org/hawkular/apm/server/infinispan/InfinispanSpanCache.java

```diff
@@ -82,6 +88,13 @@ public Span get(String tenantId, String id) {
 
     @Override
     public void store(String tenantId, List<Span> spans) throws CacheException {
+        store(tenantId, spans, x -> x.getId());
+    }
+
+    @Override
+    public void store(String tenantId, List<Span> spans, Function<Span, String> uniqueIdGenerator)
```

Comment:
Actually, I see right above that you make a call to this method, passing a function. Perhaps this should be used, but it isn't? Or perhaps I'm not looking at the right place :)